### PR TITLE
unleash confusion map

### PR DESCRIPTION
**Pull request by pavoljuhas**
Revert "Guard confusion_map usage (#5534)"

Assume measure() methods of SimulationState subclasses accept the
`confusion_map` argument.  Remove temporary workaround for methods
not taking `confusion_map`.

Also disable pylint check on missing argument which is required
in the base class but unknown to subclasses actually called.

This reverts commit 25388af7d96edb88ef320a08374e8b9b871126b3.

### CODE REVIEW

File: cirq-core/cirq/sim/simulation_state.py [hunk 1]
```diff
@@ -187,7 +186,11 @@ def with_qubits(self: TSelf, qubits) -> TSelf:
         Regurns:
```
Comment (95-martin-orion) [inline cirq-core/cirq/sim/simulation_state.py:186]: While we're here: `Regurns` -> `Returns`

File: cirq-core/cirq/sim/simulation_state.py [hunk 2]
```diff
@@ -187,7 +186,11 @@ def with_qubits(self: TSelf, qubits) -> TSelf:
...
             A new subclass object containing the extended state space.
         """
+        # FIXME: SimulationState.__init__ requires the `state` argument, but
```
Comment (95-martin-orion) [inline cirq-core/cirq/sim/simulation_state.py:189]: Could you open an issue for this and reference it here?

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]
Excellent PR naming.

File: cirq-core/cirq/sim/simulation_state.py (hunk 2 above)
Comment (pavoljuhas) [reply cirq-core/cirq/sim/simulation_state.py]: Done - opened #5721 and added a TODO link to that issue.

**Review by 95-martin-orion — APPROVED** [review]

**Pull request merged**